internal/driver/config_test.go

```diff
@@ -51,3 +54,48 @@ func TestUpdateFromRaw(t *testing.T) {
 		})
 	}
 }
+
+// TestGetCameraXAddr verify the parsing of the camera XAddr
+func TestGetCameraXAddr(t *testing.T) {
+
+	tests := []struct {
+		input    map[string]models.ProtocolProperties
+		expected string
+
+		errorExpected bool
+	}{
+		{
+			input: map[string]models.ProtocolProperties{
+				OnvifProtocol: {
+					Address: "http://localhost",
+					Port:    "80",
+				},
+			},
+			expected: "http://localhost:80",
+		},
+		{
+			input: map[string]models.ProtocolProperties{
+				OnvifProtocol: {
+					CustomMetadata: "custommetadata",
+				},
+			},
+			errorExpected: true,
+		},
+		{
+			errorExpected: true,
+		},
+	}
```

Comment:
One more test case that might be useful to test is passing empty string for address and checking for error.